ambari-server/src/main/java/org/apache/ambari/server/topology/HostGroup.java

```diff
@@ -79,35 +81,38 @@
    * @return collection of component names as String that are associated with
    *           the specified provision action
    */
+  @Deprecated
```

Comment:
Why is this method deprecated?

Reply:
Eventually I think we'll need to pass Component objects (which can contain information about mpack and service instance as needed) as passing simply the component name could be ambiguous in the multi-everything world.